x-pack/plugins/apm/public/components/app/service_overview/service_overview_instances_table/instance_actions_menu/menu_sections.ts

```diff
@@ -0,0 +1,203 @@
+        { defaultMessage: 'Container logs' }
+      ),
+      href: getInfraHref({
+        app: 'logs',
+        basePath,
+        path: `/link-to/container-logs/${containerId}`,
+        query: { time },
+      }),
+      condition: !!containerId,
+    },
+    {
+      key: 'containerMetrics',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.containerMetrics',
+        { defaultMessage: 'Container metrics' }
+      ),
+      href: getInfraHref({
+        app: 'metrics',
+        basePath,
+        path: `/link-to/container-detail/${containerId}`,
+        query: infraMetricsQuery,
+      }),
+      condition: !!containerId,
+    },
+  ];
+
+  const apmActions: Action[] = [
+    {
+      key: 'filterByInstance',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.filterByInstance',
+        {
+          defaultMessage: 'Filter overview by instance',
+        }
+      ),
+      onClick: onFilterByInstanceClick,
+      condition: true,
+    },
+    {
+      key: 'analyzeRuntimeMetric',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.metrics',
+        {
+          defaultMessage: 'Metrics',
+        }
+      ),
+      href: metricsHref,
+      condition: true,
+    },
+  ];
+
+  const sectionRecord: SectionRecord = {
+    observability: [
+      {
+        key: 'podDetails',
+        title: i18n.translate(
+          'xpack.apm.serviceOverview.instancesTable.actionMenus.pod.title',
+          {
+            defaultMessage: 'Pod details',
+          }
+        ),
+        subtitle: i18n.translate(
+          'xpack.apm.serviceOverview.instancesTable.actionMenus.pod.subtitle',
+          {
+            defaultMessage:
+              'View logs and metrics for this pod to get further details.',
+          }
+        ),
+        actions: podActions,
+      },
+      {
+        key: 'containerDetails',
+        title: i18n.translate(
+          'xpack.apm.serviceOverview.instancesTable.actionMenus.container.title',
+          {
+            defaultMessage: 'Container details',
+          }
+        ),
+        subtitle: i18n.translate(
+          'xpack.apm.serviceOverview.instancesTable.actionMenus.container.subtitle',
+          {
+            defaultMessage:
+              'View logs and metrics for this container to get further details.',
+          }
+        ),
+        actions: containerActions,
+      },
+    ],
+    apm: [{ key: 'apm', actions: apmActions }],
+  };
+
+  // Filter out actions that shouldnt be shown and sections without any actions.
+  return Object.values(sectionRecord)
+    .map((sections) =>
+      sections
+        .map((section) => ({
+          ...section,
+          actions: section.actions.filter((action) => action.condition),
+        }))
+        .filter((section) => !isEmpty(section.actions))
+    )
+    .filter((sections) => !isEmpty(sections));
+}
```

Comment:
@cauemarcondes This looks like duplication from https://github.com/elastic/kibana/blob/5d95e2e0cd59438e76e7a585eed7ac382f78b057/x-pack/plugins/apm/public/components/shared/TransactionActionMenu/sections.ts

Can we avoid that?

Reply:
yes you are right, I refactored it here https://github.com/elastic/kibana/pull/104338